source/server/connection_handler_impl.cc

```diff
@@ -39,60 +39,86 @@ void ConnectionHandlerImpl::addListener(absl::optional<uint64_t> overridden_list
     return;
   }
 
-  ActiveListenerDetails details;
+  auto details = std::make_shared<ActiveListenerDetails>();
   if (config.internalListenerConfig().has_value()) {
     if (overridden_listener.has_value()) {
-      for (auto& listener : listeners_) {
-        if (listener.second.listener_->listenerTag() == overridden_listener) {
-          listener.second.internalListener()->get().updateListenerConfig(config);
-          return;
-        }
+      auto listener_it = listener_map_by_tag_.find(overridden_listener.value());
+      if (listener_it != listener_map_by_tag_.end()) {
+        listener_it->second->internalListener()->get().updateListenerConfig(config);
+        return;
```

Comment:
nit:
```
if(auto iter = listener_map_by_tag_.find(overridden_listener.value()); iter != listener_map_by_tag_.end()) {
  // ...
}
```

Reply:
Personal preference and feel free to just keep original code.

Reply:
done